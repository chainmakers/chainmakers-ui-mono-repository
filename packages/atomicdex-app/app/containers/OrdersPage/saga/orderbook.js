// @flow
/* eslint no-param-reassign: ["error", { "props": false }] */
import { put, call, select, cancelled } from 'redux-saga/effects';
import { CANCEL, delay } from 'redux-saga';
import api from 'utils/barterdex-api';
import { openSnackbars } from '../../Snackbars/actions';
import { makeSelectBalanceEntities } from '../../App/selectors';
import { ORDER_BOB_SIDE, ORDER_ALICE_SIDE } from '../constants';
import {
  makeSelectOrderbookDeposit,
  makeSelectOrderbookRecevie
} from '../selectors';
import {
  reloadOrderbookSuccess,
  loadOrderbookSuccess,
  loadOrderbookError,
  skipOrderbook
} from '../actions';

const debug = require('debug')(
  'atomicapp:containers:OrdersPage:saga:orderbook'
);

export function* listenForReloadingOrderbook(action, timeout = 60) {
  debug(`listen for reloading orderbook`);
  let request = null;
  try {
    // if not found stop
    const deposit = yield select(makeSelectOrderbookDeposit());
    const recevie = yield select(makeSelectOrderbookRecevie());
    const balance = yield select(makeSelectBalanceEntities());
    const address = balance.get(deposit).get('address');

    if (!deposit || !recevie || deposit === recevie) {
      return yield put(skipOrderbook());
    }
    let n = timeout; // 60s
    while (true) {
      n -= 1;
      debug('start');
      if (n < 0) {
        debug('stop, timeout');
        break;
      }
      request = api.orderbook({
        base: recevie,
        rel: deposit,
        queueid: undefined
      });
      const result = yield request;
      result.bids.map(v => {
        v.base = recevie;
        v.rel = deposit;
        v.id = `${v.address}-${deposit}-${recevie}`;
        v.type = ORDER_ALICE_SIDE;
        return v;
      });
      result.asks.map(v => {
        v.base = recevie;
        v.rel = deposit;
        v.id = `${v.address}-${deposit}-${recevie}`;
        v.type = ORDER_BOB_SIDE;
        return v;
      });
      yield put(loadOrderbookSuccess(result));
      const f = result.bids.find(e => e.address === address);
      if (f && f.maxvolume > 0) {
        break;
      }
      yield call(delay, 1000);
    }
    yield put(reloadOrderbookSuccess());
  } catch (err) {
    debug(`reloading orderbook error: ${err.message}`);
    yield put(openSnackbars(err.message));
    yield put(
      loadOrderbookError({
        context: {
          action: action.type
        },
        type: 'RPC',
        message: err.message
      })
    );
  } finally {
    if (yield cancelled()) {
      debug(`reloading orderbook cancelled`);
      if (request && request[CANCEL]) {
        request[CANCEL]();
      }
    }
  }
}

export default function* listenForLoadingOrderbook(action) {
  debug(`listen for loading orderbook`);

  let request = null;
  try {
    const deposit = yield select(makeSelectOrderbookDeposit());
    const recevie = yield select(makeSelectOrderbookRecevie());

    if (!deposit || !recevie) {
      return yield put(skipOrderbook());
    }

    request = api.orderbook({
      base: recevie,
      rel: deposit,
      queueid: undefined
    });
    const result = yield request;
    result.bids.map(v => {
      v.base = recevie;
      v.rel = deposit;
      v.id = `${v.address}-${deposit}-${recevie}`;
      v.type = ORDER_ALICE_SIDE;
      return v;
    });
    result.asks.map(v => {
      v.base = recevie;
      v.rel = deposit;
      v.id = `${v.address}-${deposit}-${recevie}`;
      v.type = ORDER_BOB_SIDE;
      return v;
    });
    return yield put(loadOrderbookSuccess(result));
  } catch (err) {
    debug(`loading orderbook error: ${err.message}`);
    yield put(openSnackbars(err.message));
    yield put(
      loadOrderbookError({
        context: {
          action: action.type
        },
        type: 'RPC',
        message: err.message
      })
    );
  } finally {
    if (yield cancelled()) {
      debug(`loading orderbook cancelled`);
      if (request && request[CANCEL]) {
        request[CANCEL]();
      }
    }
  }
}
