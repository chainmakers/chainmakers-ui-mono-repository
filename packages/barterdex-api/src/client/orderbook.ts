// @doc
// https://docs.komodoplatform.com/barterDEX/barterDEX-API.html#orderbook

interface OrderbookType {
  base: string,
  rel: string
};

export default function orderbookFactory() {
  return {
    orderbook(params: OrderbookType) {
      const serverparams = Object.assign({}, params, {
        method: 'orderbook'
      });
      return this.privateCall(serverparams);
    }
  };
}
