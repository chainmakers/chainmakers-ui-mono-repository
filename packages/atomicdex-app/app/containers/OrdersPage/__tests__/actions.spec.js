import orderbook from '../../__tests__/orderbook.json';
import {
  loadPrice,
  loadPrices,
  loadPricesSuccess,
  loadBestPrice,
  loadBuyCoin,
  loadBuyCoinError,
  checkTimeoutEvent,
  checkUpdateSwapEvent,
  timeoutSwap,
  makeANewSwap,
  openDetailModal,
  closeDetailModal,
  openJoyride,
  closeJoyride,
  loadOrderbook,
  skipOrderbook,
  loadOrderbookSuccess,
  loadOrderbookError,
  openDepositCoinModal,
  closeDepositCoinModal,
  selectCoinDeposit,
  openRecevieCoinModal,
  closeRecevieCoinModal,
  selectCoinRecevie,
  setNewOrder,
  skipNewOrder,
  setNewOrderSuccess,
  setNewOrderError,
  openConfirmNewOrderModal,
  closeConfirmNewOrderModal
} from '../actions';

import {
  LOAD_PRICE,
  LOAD_PRICES,
  LOAD_PRICES_SUCCESS,
  LOAD_BEST_PRICE,
  LOAD_BUY_COIN,
  LOAD_BUY_COIN_ERROR,
  CHECK_TIMEOUT_EVENT,
  CHECK_UPDATE_SWAP_EVENT,
  SWAP_TIMEOUT,
  SWAP_MAKE_A_NEW,
  SWAP_DETAIL_MODAL_OPEN,
  SWAP_DETAIL_MODAL_CLOSE,
  JOYRIDE_OPEN,
  JOYRIDE_CLOSE,
  ORDERBOOK_LOAD,
  ORDERBOOK_LOAD_SKIP,
  ORDERBOOK_LOAD_SUCCESS,
  ORDERBOOK_LOAD_ERROR,
  DEPOSIT_COIN_MODAL_OPEN,
  DEPOSIT_COIN_MODAL_CLOSE,
  RECEVIE_COIN_MODAL_OPEN,
  RECEVIE_COIN_MODAL_CLOSE,
  NEW_ORDER_SET,
  NEW_ORDER_SET_SKIP,
  NEW_ORDER_SET_SUCCESS,
  NEW_ORDER_SET_ERROR,
  CONFIRM_NEW_ORDER_MODAL_OPEN,
  CONFIRM_NEW_ORDER_MODAL_CLOSE
} from '../constants';

describe('containers/DexPage/actions/loadPrice', () => {
  const coin = 'coin';

  it('should loadPrice should create loadPrice action', () => {
    expect(loadPrice(coin)).toMatchSnapshot();
  });

  it('should return the correct type and the passed name', () => {
    const expectedResult = {
      type: LOAD_PRICE,
      payload: {
        coin
      }
    };

    expect(loadPrice(coin)).toEqual(expectedResult);
  });
});

describe('containers/DexPage/actions/loadPrices', () => {
  it('should loadPrices should create loadPrices action', () => {
    expect(loadPrices()).toMatchSnapshot();
  });

  it('should return the correct type and the passed name', () => {
    const expectedResult = {
      type: LOAD_PRICES
    };

    expect(loadPrices()).toEqual(expectedResult);
  });
});

describe('containers/DexPage/actions/loadPricesSuccess', () => {
  it('should loadPricesSuccess should create loadPricesSuccess action', () => {
    expect(loadPricesSuccess()).toMatchSnapshot();
  });

  it('should return the correct type and the passed name', () => {
    const expectedResult = {
      type: LOAD_PRICES_SUCCESS
    };

    expect(loadPricesSuccess()).toEqual(expectedResult);
  });
});

describe('containers/DexPage/actions/loadBestPrice', () => {
  const payload = {
    address: '',
    age: 0,
    avevolume: 0,
    base: 'COQUI',
    bestPrice: 0,
    depth: 0,
    maxvolume: 0,
    numutxos: 0,
    price: 0,
    pubkey: '',
    rel: 'PIZZA',
    zcredits: 0
  };
  it('should loadBestPrice should create loadBestPrice action', () => {
    expect(loadBestPrice(payload)).toMatchSnapshot();
  });

  it('should return the correct type and the passed name', () => {
    const expectedResult = {
      type: LOAD_BEST_PRICE,
      payload
    };

    expect(loadBestPrice(payload)).toEqual(expectedResult);
  });
});

describe('containers/DexPage/actions/loadBuyCoin', () => {
  const payload = {
    basecoin: 'BTC',
    paymentcoin: 'KMD',
    amount: 1.234
  };

  it('should loadBuyCoin should create loadBuyCoin action', () => {
    expect(loadBuyCoin(payload)).toMatchSnapshot();
  });

  it('should return the correct type and the passed name', () => {
    const expectedResult = {
      type: LOAD_BUY_COIN,
      payload
    };

    expect(loadBuyCoin(payload)).toEqual(expectedResult);
  });
});

describe('containers/DexPage/actions/loadBuyCoinError', () => {
  const message = 'Value is large than balance';

  it('should loadBuyCoin should create loadBuyCoinError action', () => {
    expect(loadBuyCoinError(message)).toMatchSnapshot();
  });

  it('should return the correct type and the passed name', () => {
    const expectedResult = {
      type: LOAD_BUY_COIN_ERROR,
      error: {
        message
      }
    };

    expect(loadBuyCoinError(message)).toEqual(expectedResult);
  });
});

describe('containers/DexPage/actions/checkTimeoutEvent', () => {
  it('should checkTimeoutEvent should create checkTimeoutEvent action', () => {
    expect(checkTimeoutEvent()).toMatchSnapshot();
  });

  it('should return the correct type and the passed name', () => {
    const expectedResult = {
      type: CHECK_TIMEOUT_EVENT
    };

    expect(checkTimeoutEvent()).toEqual(expectedResult);
  });
});

describe('containers/DexPage/actions/checkUpdateSwapEvent', () => {
  it('should checkUpdateSwapEvent should create checkUpdateSwapEvent action', () => {
    expect(checkUpdateSwapEvent()).toMatchSnapshot();
  });

  it('should return the correct type and the passed name', () => {
    const expectedResult = {
      type: CHECK_UPDATE_SWAP_EVENT
    };

    expect(checkUpdateSwapEvent()).toEqual(expectedResult);
  });
});

describe('containers/DexPage/actions/timeoutSwap', () => {
  const payload = {
    alice: 'BEER',
    bob: 'COQUI',
    id: 3624682363,
    quoteid: 0,
    requestid: 0,
    uuid: 'bc5e1509b2aea898b8dff71ecc3fa7d5bc7c361fb14187fe9bc06916fae63811'
  };
  it('should timeoutSwap should create timeoutSwap action', () => {
    expect(timeoutSwap(payload)).toMatchSnapshot();
  });

  it('should return the correct type and the passed name', () => {
    const expectedResult = {
      type: SWAP_TIMEOUT,
      payload
    };

    expect(timeoutSwap(payload)).toEqual(expectedResult);
  });
});

describe('containers/DexPage/actions/makeANewSwap', () => {
  it('should makeANewSwap should create makeANewSwap action', () => {
    expect(makeANewSwap()).toMatchSnapshot();
  });

  it('should return the correct type and the passed name', () => {
    const expectedResult = {
      type: SWAP_MAKE_A_NEW
    };

    expect(makeANewSwap()).toEqual(expectedResult);
  });
});

describe('containers/DexPage/actions/openDetailModal', () => {
  const uuid = 'uuid';
  it('should openDetailModal should create openDetailModal action', () => {
    expect(openDetailModal()).toMatchSnapshot();
    expect(openDetailModal(uuid)).toMatchSnapshot();
  });

  it('should return the correct type', () => {
    const expectedResult = {
      type: SWAP_DETAIL_MODAL_OPEN,
      payload: {
        uuid: undefined
      }
    };

    expect(openDetailModal()).toEqual(expectedResult);
  });

  it('should return the correct type and the uuid', () => {
    const expectedResult = {
      type: SWAP_DETAIL_MODAL_OPEN,
      payload: {
        uuid
      }
    };

    expect(openDetailModal(uuid)).toEqual(expectedResult);
  });
});

describe('containers/DexPage/actions/closeDetailModal', () => {
  it('should closeDetailModal should create closeDetailModal action', () => {
    expect(closeDetailModal()).toMatchSnapshot();
  });

  it('should return the correct type and the passed name', () => {
    const expectedResult = {
      type: SWAP_DETAIL_MODAL_CLOSE
    };

    expect(closeDetailModal()).toEqual(expectedResult);
  });
});

describe('containers/DexPage/actions/openJoyride', () => {
  it('should openJoyride should create openJoyride action', () => {
    expect(openJoyride()).toMatchSnapshot();
  });

  it('should return the correct type and the passed name', () => {
    const expectedResult = {
      type: JOYRIDE_OPEN
    };

    expect(openJoyride()).toEqual(expectedResult);
  });
});

describe('containers/DexPage/actions/closeJoyride', () => {
  it('should closeJoyride should create closeJoyride action', () => {
    expect(closeJoyride()).toMatchSnapshot();
  });

  it('should return the correct type and the passed name', () => {
    const expectedResult = {
      type: JOYRIDE_CLOSE
    };

    expect(closeJoyride()).toEqual(expectedResult);
  });
});

describe('containers/OrderPage/actions/loadOrderbook', () => {
  it('should loadOrderbook should create loadOrderbook action', () => {
    expect(loadOrderbook()).toMatchSnapshot();
  });

  it('should return the correct type and the passed name', () => {
    const expectedResult = {
      type: ORDERBOOK_LOAD
    };

    expect(loadOrderbook()).toEqual(expectedResult);
  });
});

describe('containers/OrderPage/actions/skipOrderbook', () => {
  it('should skipOrderbook should create skipOrderbook action', () => {
    expect(skipOrderbook()).toMatchSnapshot();
  });

  it('should return the correct type and the passed name', () => {
    const expectedResult = {
      type: ORDERBOOK_LOAD_SKIP
    };

    expect(skipOrderbook()).toEqual(expectedResult);
  });
});

describe('containers/OrderPage/actions/loadOrderbookSuccess', () => {
  const payload = orderbook;

  it('should loadOrderbookSuccess should create loadOrderbookSuccess action', () => {
    expect(loadOrderbookSuccess(payload)).toMatchSnapshot();
  });

  it('should return the correct type and the passed name', () => {
    const expectedResult = {
      type: ORDERBOOK_LOAD_SUCCESS,
      payload
    };

    expect(loadOrderbookSuccess(payload)).toEqual(expectedResult);
  });
});

describe('containers/OrderPage/actions/loadOrderbookError', () => {
  const error = {
    context: {
      action: ORDERBOOK_LOAD
    },
    message: 'message',
    type: 'RPC'
  };

  it('should loadOrderbookError should create loadOrderbookError action', () => {
    expect(loadOrderbookError(error)).toMatchSnapshot();
  });

  it('should return the correct type and the passed name', () => {
    const expectedResult = {
      type: ORDERBOOK_LOAD_ERROR,
      error
    };

    expect(loadOrderbookError(error)).toEqual(expectedResult);
  });
});

describe('containers/OrdersPage/actions/openDepositCoinModal', () => {
  it('should openDepositCoinModal should create openDepositCoinModal action', () => {
    expect(openDepositCoinModal()).toMatchSnapshot();
  });

  it('should return the correct type and the passed name', () => {
    const expectedResult = {
      type: DEPOSIT_COIN_MODAL_OPEN
    };

    expect(openDepositCoinModal()).toEqual(expectedResult);
  });
});

describe('containers/OrdersPage/actions/closeDepositCoinModal', () => {
  it('should closeDepositCoinModal should create closeDepositCoinModal action', () => {
    expect(closeDepositCoinModal()).toMatchSnapshot();
  });

  it('should return the correct type and the passed name', () => {
    const expectedResult = {
      type: DEPOSIT_COIN_MODAL_CLOSE
    };

    expect(closeDepositCoinModal()).toEqual(expectedResult);
  });
});

describe('containers/OrdersPage/actions/openRecevieCoinModal', () => {
  it('should openRecevieCoinModal should create openRecevieCoinModal action', () => {
    expect(openRecevieCoinModal()).toMatchSnapshot();
  });

  it('should return the correct type and the passed name', () => {
    const expectedResult = {
      type: RECEVIE_COIN_MODAL_OPEN
    };

    expect(openRecevieCoinModal()).toEqual(expectedResult);
  });
});

describe('containers/OrdersPage/actions/closeRecevieCoinModal', () => {
  it('should closeRecevieCoinModal should create closeRecevieCoinModal action', () => {
    expect(closeRecevieCoinModal()).toMatchSnapshot();
  });

  it('should return the correct type and the passed name', () => {
    const expectedResult = {
      type: RECEVIE_COIN_MODAL_CLOSE
    };

    expect(closeRecevieCoinModal()).toEqual(expectedResult);
  });
});

describe('containers/OrdersPage/actions/setNewOrder', () => {
  const price = 0.1;
  it('should setNewOrder should create setNewOrder action', () => {
    expect(setNewOrder(price)).toMatchSnapshot();
  });

  it('should return the correct type and the passed name', () => {
    const expectedResult = {
      type: NEW_ORDER_SET,
      payload: {
        price
      }
    };

    expect(setNewOrder(price)).toEqual(expectedResult);
  });
});

describe('containers/OrdersPage/actions/setNewOrderSuccess', () => {
  it('should setNewOrderSuccess should create setNewOrderSuccess action', () => {
    expect(setNewOrderSuccess()).toMatchSnapshot();
  });

  it('should return the correct type and the passed name', () => {
    const expectedResult = {
      type: NEW_ORDER_SET_SUCCESS
    };

    expect(setNewOrderSuccess()).toEqual(expectedResult);
  });
});

describe('containers/OrdersPage/actions/setNewOrderError', () => {
  it('should setNewOrderError should create setNewOrderError action', () => {
    expect(setNewOrderError()).toMatchSnapshot();
  });

  it('should return the correct type and the passed name', () => {
    const expectedResult = {
      type: NEW_ORDER_SET_ERROR
    };

    expect(setNewOrderError()).toEqual(expectedResult);
  });
});

describe('containers/OrdersPage/actions/skipNewOrder', () => {
  it('should skipNewOrder should create skipNewOrder action', () => {
    expect(skipNewOrder()).toMatchSnapshot();
  });

  it('should return the correct type and the passed name', () => {
    const expectedResult = {
      type: NEW_ORDER_SET_SKIP
    };

    expect(skipNewOrder()).toEqual(expectedResult);
  });
});

describe('containers/OrdersPage/actions/openConfirmNewOrderModal', () => {
  it('should openConfirmNewOrderModal should create openConfirmNewOrderModal action', () => {
    expect(openConfirmNewOrderModal()).toMatchSnapshot();
  });

  it('should return the correct type and the passed name', () => {
    const expectedResult = {
      type: CONFIRM_NEW_ORDER_MODAL_OPEN
    };

    expect(openConfirmNewOrderModal()).toEqual(expectedResult);
  });
});

describe('containers/OrdersPage/actions/closeConfirmNewOrderModal', () => {
  it('should closeConfirmNewOrderModal should create closeConfirmNewOrderModal action', () => {
    expect(closeConfirmNewOrderModal()).toMatchSnapshot();
  });

  it('should return the correct type and the passed name', () => {
    const expectedResult = {
      type: CONFIRM_NEW_ORDER_MODAL_CLOSE
    };

    expect(closeConfirmNewOrderModal()).toEqual(expectedResult);
  });
});
