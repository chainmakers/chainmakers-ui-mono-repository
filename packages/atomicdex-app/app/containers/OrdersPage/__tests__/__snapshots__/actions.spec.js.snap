// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/DexPage/actions/checkTimeoutEvent should checkTimeoutEvent should create checkTimeoutEvent action 1`] = `
Object {
  "type": "atomicapp/DexPage/CHECK_TIMEOUT_EVENT",
}
`;

exports[`containers/DexPage/actions/checkUpdateSwapEvent should checkUpdateSwapEvent should create checkUpdateSwapEvent action 1`] = `
Object {
  "type": "atomicapp/DexPage/CHECK_UPDATE_SWAP_EVENT",
}
`;

exports[`containers/DexPage/actions/closeDetailModal should closeDetailModal should create closeDetailModal action 1`] = `
Object {
  "type": "atomicapp/DexPage/SWAP_DETAIL_MODAL_CLOSE",
}
`;

exports[`containers/DexPage/actions/closeJoyride should closeJoyride should create closeJoyride action 1`] = `
Object {
  "type": "atomicapp/DexPage/JOYRIDE_CLOSE",
}
`;

exports[`containers/DexPage/actions/loadBestPrice should loadBestPrice should create loadBestPrice action 1`] = `
Object {
  "payload": Object {
    "address": "",
    "age": 0,
    "avevolume": 0,
    "base": "COQUI",
    "bestPrice": 0,
    "depth": 0,
    "maxvolume": 0,
    "numutxos": 0,
    "price": 0,
    "pubkey": "",
    "rel": "PIZZA",
    "zcredits": 0,
  },
  "type": "atomicapp/DexPage/LOAD_BEST_PRICE",
}
`;

exports[`containers/DexPage/actions/loadBuyCoin should loadBuyCoin should create loadBuyCoin action 1`] = `
Object {
  "payload": Object {
    "amount": 1.234,
    "basecoin": "BTC",
    "paymentcoin": "KMD",
  },
  "type": "atomicapp/DexPage/LOAD_BUY_COIN",
}
`;

exports[`containers/DexPage/actions/loadBuyCoinError should loadBuyCoin should create loadBuyCoinError action 1`] = `
Object {
  "error": Object {
    "message": "Value is large than balance",
  },
  "type": "atomicapp/DexPage/LOAD_BUY_COIN_ERROR",
}
`;

exports[`containers/DexPage/actions/loadPrice should loadPrice should create loadPrice action 1`] = `
Object {
  "payload": Object {
    "coin": "coin",
  },
  "type": "atomicapp/DexPage/LOAD_PRICE",
}
`;

exports[`containers/DexPage/actions/loadPrices should loadPrices should create loadPrices action 1`] = `
Object {
  "type": "atomicapp/DexPage/LOAD_PRICES",
}
`;

exports[`containers/DexPage/actions/loadPricesSuccess should loadPricesSuccess should create loadPricesSuccess action 1`] = `
Object {
  "type": "atomicapp/DexPage/LOAD_PRICES_SUCCESS",
}
`;

exports[`containers/DexPage/actions/makeANewSwap should makeANewSwap should create makeANewSwap action 1`] = `
Object {
  "type": "atomicapp/DexPage/SWAP_MAKE_A_NEW",
}
`;

exports[`containers/DexPage/actions/openDetailModal should openDetailModal should create openDetailModal action 1`] = `
Object {
  "payload": Object {
    "uuid": undefined,
  },
  "type": "atomicapp/DexPage/SWAP_DETAIL_MODAL_OPEN",
}
`;

exports[`containers/DexPage/actions/openDetailModal should openDetailModal should create openDetailModal action 2`] = `
Object {
  "payload": Object {
    "uuid": "uuid",
  },
  "type": "atomicapp/DexPage/SWAP_DETAIL_MODAL_OPEN",
}
`;

exports[`containers/DexPage/actions/openJoyride should openJoyride should create openJoyride action 1`] = `
Object {
  "type": "atomicapp/DexPage/JOYRIDE_OPEN",
}
`;

exports[`containers/DexPage/actions/timeoutSwap should timeoutSwap should create timeoutSwap action 1`] = `
Object {
  "payload": Object {
    "alice": "BEER",
    "bob": "COQUI",
    "id": 3624682363,
    "quoteid": 0,
    "requestid": 0,
    "uuid": "bc5e1509b2aea898b8dff71ecc3fa7d5bc7c361fb14187fe9bc06916fae63811",
  },
  "type": "atomicapp/DexPage/SWAP_TIMEOUT",
}
`;

exports[`containers/OrderPage/actions/loadOrderbook should loadOrderbook should create loadOrderbook action 1`] = `
Object {
  "type": "atomicapp/OrdersPage/ORDERBOOK_LOAD",
}
`;

exports[`containers/OrderPage/actions/loadOrderbookError should loadOrderbookError should create loadOrderbookError action 1`] = `
Object {
  "error": Object {
    "context": Object {
      "action": "atomicapp/OrdersPage/ORDERBOOK_LOAD",
    },
    "message": "message",
    "type": "RPC",
  },
  "type": "atomicapp/OrdersPage/ORDERBOOK_LOAD_ERROR",
}
`;

exports[`containers/OrderPage/actions/loadOrderbookSuccess should loadOrderbookSuccess should create loadOrderbookSuccess action 1`] = `
Object {
  "payload": Object {
    "askdepth": 0,
    "asks": Array [
      Object {
        "address": "RT9MpMyucqXiX8bZLimXBnrrn2ofmdGNKd",
        "age": 9,
        "avevolume": 0,
        "coin": "KMD",
        "depth": 0,
        "maxvolume": 36.40686108,
        "numutxos": 0,
        "price": 0.0001566,
        "pubkey": "bab6ad2eebe1e666369cab504d4622b22c1f1ef718ef388e88020f30a1573e01",
        "zcredits": 0,
      },
    ],
    "base": "KMD",
    "biddepth": 0,
    "bids": Array [
      Object {
        "address": "1JsAjr6d21j9T8EMsYnQ6GXf1mM523JAv1",
        "age": 9,
        "avevolume": 0,
        "coin": "BTC",
        "depth": 0,
        "maxvolume": 0.02620853,
        "numutxos": 0,
        "price": 0.00015473,
        "pubkey": "bab6ad2eebe1e666369cab504d4622b22c1f1ef718ef388e88020f30a1573e01",
        "zcredits": 0,
      },
    ],
    "netid": 9999,
    "numasks": 1,
    "numbids": 1,
    "rel": "BTC",
    "timestamp": 1558682258,
  },
  "type": "atomicapp/OrdersPage/ORDERBOOK_LOAD_SUCCESS",
}
`;

exports[`containers/OrderPage/actions/skipOrderbook should skipOrderbook should create skipOrderbook action 1`] = `
Object {
  "type": "atomicapp/OrdersPage/ORDERBOOK_LOAD_SKIP",
}
`;

exports[`containers/OrdersPage/actions/closeConfirmNewOrderModal should closeConfirmNewOrderModal should create closeConfirmNewOrderModal action 1`] = `
Object {
  "type": "atomicapp/OrdersPage/CONFIRM_NEW_ORDER_MODAL_CLOSE",
}
`;

exports[`containers/OrdersPage/actions/closeDepositCoinModal should closeDepositCoinModal should create closeDepositCoinModal action 1`] = `
Object {
  "type": "atomicapp/OrdersPage/DEPOSIT_COIN_MODAL_CLOSE",
}
`;

exports[`containers/OrdersPage/actions/closeRecevieCoinModal should closeRecevieCoinModal should create closeRecevieCoinModal action 1`] = `
Object {
  "type": "atomicapp/OrdersPage/RECEVIE_COIN_MODAL_CLOSE",
}
`;

exports[`containers/OrdersPage/actions/openConfirmNewOrderModal should openConfirmNewOrderModal should create openConfirmNewOrderModal action 1`] = `
Object {
  "type": "atomicapp/OrdersPage/CONFIRM_NEW_ORDER_MODAL_OPEN",
}
`;

exports[`containers/OrdersPage/actions/openDepositCoinModal should openDepositCoinModal should create openDepositCoinModal action 1`] = `
Object {
  "type": "atomicapp/OrdersPage/DEPOSIT_COIN_MODAL_OPEN",
}
`;

exports[`containers/OrdersPage/actions/openRecevieCoinModal should openRecevieCoinModal should create openRecevieCoinModal action 1`] = `
Object {
  "type": "atomicapp/OrdersPage/RECEVIE_COIN_MODAL_OPEN",
}
`;

exports[`containers/OrdersPage/actions/setNewOrder should setNewOrder should create setNewOrder action 1`] = `
Object {
  "payload": Object {
    "price": 0.1,
  },
  "type": "atomicapp/OrdersPage/NEW_ORDER_SET",
}
`;

exports[`containers/OrdersPage/actions/setNewOrderError should setNewOrderError should create setNewOrderError action 1`] = `
Object {
  "error": undefined,
  "type": "atomicapp/OrdersPage/NEW_ORDER_SET_ERROR",
}
`;

exports[`containers/OrdersPage/actions/setNewOrderSuccess should setNewOrderSuccess should create setNewOrderSuccess action 1`] = `
Object {
  "type": "atomicapp/OrdersPage/NEW_ORDER_SET_SUCCESS",
}
`;

exports[`containers/OrdersPage/actions/skipNewOrder should skipNewOrder should create skipNewOrder action 1`] = `
Object {
  "type": "atomicapp/OrdersPage/NEW_ORDER_SET_SKIP",
}
`;
